src/components/App.test.jsx

```diff
@@ -0,0 +1,122 @@
+import React from 'react'
+import { render, screen, waitForElementToBeRemoved } from '@testing-library/react'
+import { act } from 'react-dom/test-utils'
+import user from '@testing-library/user-event'
+
+import { BrowserRouter } from 'react-router-dom'
+
+import { SettingsProvider } from '../context/SettingsContext'
+import { WalletProvider } from '../context/WalletContext'
+import { WebsocketProvider } from '../context/WebsocketContext'
+
+import App from './App'
+
+const AllTheProviders = ({ children }) => {
+  return (
+    <BrowserRouter>
+      <SettingsProvider>
+        <WalletProvider>
+          <WebsocketProvider>{children}</WebsocketProvider>
+        </WalletProvider>
+      </SettingsProvider>
+    </BrowserRouter>
+  )
+}
+
+const addToAppSettings = () => {
+  global.localStorage.setItem(
+    global.JM.SETTINGS_STORE_KEY,
+    JSON.stringify(
+      Object.assign({}, global.localStorage.getItem(global.JM.SETTINGS_STORE_KEY) || {}, {
+        showOnboarding: false,
+      })
+    )
+  )
+}
+
+it('Display Onboarding screen initially', () => {
+  act(() => {
+    render(<App />, {
+      wrapper: AllTheProviders,
+    })
+  })
+
+  // Onboarding screen
+  expect(screen.getByText('Get started')).toBeInTheDocument()
+  expect(screen.getByText('Skip intro')).toBeInTheDocument()
+
+  // Wallets screen shown after Intro is skipped
+  expect(screen.queryByText('Your wallets')).not.toBeInTheDocument()
+
+  act(() => {
+    const skipIntro = screen.getByText('Skip intro')
+    user.click(skipIntro)
+  })
+
+  expect(screen.getByText('Your wallets')).toBeInTheDocument()
+})
+
+it('Display Wallets screen after Onboarding screen', () => {
```

Comment:
```suggestion
it('Should display Wallets directly screen when Onboarding has been shown already', () => {
```

or something like this is what you're testing here right?